include/drivers/can.h

```diff
@@ -293,6 +340,8 @@ typedef enum can_state (*can_get_state_t)(const struct device *dev,
 typedef void(*can_register_state_change_isr_t)(const struct device *dev,
 					       can_state_change_isr_t isr);
 
+typedef uint32_t (*can_get_core_clock_t)(const struct device *dev);
```

Comment:
```suggestion
typedef int (*can_get_core_clock_t)(const struct device *dev, uint32_t *rate);
```

Being able to return an error is important. It may be that communication with an external CAN controller for reading the clock rate failed, but with the proposed function signature, we have no way of propagating this (indeed, some of the drivers in this PR simply ignore this an return `0` in case of failure).

Reply:
True. It makes sense to use the same signature as clock_control_get_rate.